src/core/hle/service/ldr/ldr.cpp

```diff
@@ -39,47 +39,68 @@ constexpr ResultCode ERROR_NOT_INITIALIZED{ErrorModule::Loader, 87};
 constexpr std::size_t MAXIMUM_LOADED_RO{0x40};
 constexpr std::size_t MAXIMUM_MAP_RETRIES{0x200};
 
+struct Certification {
+    u64_le application_id_mask;
+    u64_le application_id_pattern;
+    std::array<u8, 0x10> reserved;
+    std::array<u8, 0x100> public_key; // Also known as modulus
+    std::array<u8, 0x100> signature;
+};
+static_assert(sizeof(Certification) == 0x220, "Certification has invalid size!");
+
+using SHA256Hash = std::array<u8, 0x20>;
+
 struct NRRHeader {
     u32_le magic;
-    INSERT_PADDING_BYTES(12);
-    u64_le title_id_mask;
-    u64_le title_id_pattern;
-    INSERT_PADDING_BYTES(16);
-    std::array<u8, 0x100> modulus;
-    std::array<u8, 0x100> signature_1;
-    std::array<u8, 0x100> signature_2;
-    u64_le title_id;
+    u32_le certification_signature_key_generation; // 9.0.0+
+    u64_le reserved;
+    Certification certification;
+    std::array<u8, 0x100> signature;
+    u64_le application_id;
     u32_le size;
-    INSERT_PADDING_BYTES(4);
+    u8 nrr_kind; // 7.0.0+
+    std::array<u8, 0x3> reserved_2;
     u32_le hash_offset;
     u32_le hash_count;
-    INSERT_PADDING_BYTES(8);
+    u64_le reserved_3;
+
+    // Must be dynamically allocated because, according to
+    // SwitchBrew, its size is (0x20 * hash_count) and
+    // it's impossible to determine the value of hash_count
+    // (SwitchBrew calls it "NumHash") before runtime,
+    // therefore it's not possible to calculate a SHA-256
+    std::vector<SHA256Hash> NroHashList;
 };
-static_assert(sizeof(NRRHeader) == 0x350, "NRRHeader has incorrect size.");
+
+struct SegmentHeader {
+    u32_le memory_offset;
+    u32_le memory_size;
+};
+static_assert(sizeof(SegmentHeader) == 0x8, "SegmentHeader has invalid size!");
 
 struct NROHeader {
-    INSERT_PADDING_WORDS(1);
+    // Switchbrew calls this "Start" (0x10)
+    u32_le unused;
     u32_le mod_offset;
-    INSERT_PADDING_WORDS(2);
+    u64_le padding;
+
+    // Switchbrew calls this "Header" (0x70)
     u32_le magic;
     u32_le version;
     u32_le nro_size;
     u32_le flags;
-    u32_le text_offset;
-    u32_le text_size;
-    u32_le ro_offset;
-    u32_le ro_size;
-    u32_le rw_offset;
-    u32_le rw_size;
+    // .text, .ro, .data (yuzu previously called it "rw" instead of "data")
+    std::array<SegmentHeader, 0x3> segment_headers;
```

Comment:
```suggestion
    std::array<SegmentHeader, 3> segment_headers;
```

There's also no need to reference what yuzu used to call the section.